src/Resizer/CropResizer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Resizer;
+
+use Gaufrette\File;
+use Imagine\Image\Box;
+use Imagine\Image\ImageInterface;
+use Imagine\Image\ImagineInterface;
+use Imagine\Image\Point;
+use Sonata\MediaBundle\Metadata\MetadataBuilderInterface;
+use Sonata\MediaBundle\Model\MediaInterface;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class CropResizer implements ResizerInterface
+{
+    /**
+     * @var ImagineInterface
+     */
+    private $adapter;
+
+    /**
+     * @var MetadataBuilderInterface
+     */
+    private $metadata;
+
+    public function __construct(ImagineInterface $adapter, MetadataBuilderInterface $metadata)
+    {
+        $this->adapter = $adapter;
+        $this->metadata = $metadata;
+    }
+
+    public function resize(MediaInterface $media, File $in, File $out, $format, array $settings): void
+    {
+        if (!isset($settings['width'])) {
+            throw new \RuntimeException(sprintf(
```

Comment:
I'd use `\InvalidArgumentException` instead, since this settings array is provided as argument.